app/models/user.rb

```diff
@@ -16,6 +17,27 @@ class User < ApplicationRecord
   validates :gender, presence: true
   validates :birthdate, presence: true
 
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user_name = auth.info.name.split
+      user.first_name = user_name.first
+      user.last_name = user_name.last
+      user.email = auth.info.email
```

Comment:
- [ ] **[OPTIONAL]** Instead of stubbing user information with random data, we can skip validation for gender and birthdate if we signed with facebook.

We can achieve this using `unless`:
```ruby
validates :gender, unless: -> { from_facebook? }
# same for birthdate
```
Then we can define the `from_facebook` method as private method as follows:
```ruby
def from_facebook?
  provider && uid
end
```
Because provider and uid values will not be nill in case of signing with facebook.